pkg/kubelet/network/dns/dns_test.go

```diff
@@ -121,43 +145,93 @@ func TestFormDNSSearchFitsLimits(t *testing.T) {
 	}
 
 	testCases := []struct {
-		hostNames    []string
-		resultSearch []string
-		events       []string
+		desc              string
+		hostNames         []string
+		resultSearch      []string
+		events            []string
+		expandedDNSConfig bool
 	}{
 		{
-			[]string{"testNS.svc.TEST", "svc.TEST", "TEST"},
-			[]string{"testNS.svc.TEST", "svc.TEST", "TEST"},
-			[]string{},
+			desc:         "valid: 3 search paths",
+			hostNames:    []string{"testNS.svc.TEST", "svc.TEST", "TEST"},
+			resultSearch: []string{"testNS.svc.TEST", "svc.TEST", "TEST"},
+			events:       []string{},
+		},
+
+		{
+			desc:         "valid: 5 search paths",
+			hostNames:    []string{"testNS.svc.TEST", "svc.TEST", "TEST", "AAA", "BBB"},
+			resultSearch: []string{"testNS.svc.TEST", "svc.TEST", "TEST", "AAA", "BBB"},
+			events:       []string{},
+		},
+
+		{
+			desc:         "invalid: longer than 256 characters in search path list",
+			hostNames:    []string{"testNS.svc.TEST", "svc.TEST", "TEST", "AAA", strings.Repeat("B", 256), "BBB"},
+			resultSearch: []string{"testNS.svc.TEST", "svc.TEST", "TEST", "AAA", "BBB"},
+			events:       []string{"Search Line limits were exceeded, some search paths have been omitted, the applied search line is: testNS.svc.TEST svc.TEST TEST AAA BBB"},
+		},
```

Comment:
can we add one with the offender in the middle of the list instead of being the last one?

Reply:
I think the case already has a long search path in the middle of the list.
Let me know if I miss something.

Reply:
```
hostNames:    []string{"testNS.svc.TEST", strings.Repeat("B", 256), "BBB","svc.TEST", "TEST", "AAA", },			resultSearch: []string{"testNS.svc.TEST", "svc.TEST", "TEST", "AAA", "BBB"},
```

Reply:
I am not sure what you said...
I think `strings.Repeat("B", 256)` should be filtered out and it already is the one in the middle of the list(before the "BBB").

Reply:
ahhh sorry, I totally missed that last "BBB", my apologies